### DISCUSSION TITLE
Tweak default <A-l> binding to do ls -l when cursor is on a file

### DISCUSSION THREAD

**Discussion by Tmpod** [fish-shell/fish-shell#10745] [Q&A] [answered]
Hello!

The default `<A-l>` (`Alt+L`) binding runs `ls` unless the cursor is on top of a valid directory path, in which case it runs `ls` in that directory. I'd like to extend that so that it runs `ls -l` (long listing) if the cursor is on top of a valid file path. 
Is that possible? If so, how?

Thank you!

**Comment by faho** [comment, marked as answer]
As `bind \el` will tell you:

```fish
bind --preset \el __fish_list_current_token
```

It runs `__fish_list_current_token`. And if you run `type __fish_list_current_token`, it will tell you that it's a function, so you can run `funced __fish_list_current_token` and customize it however you like.

(in the upcoming fish 4.0, you can also use `bind alt-l` instead of the escape sequence notation)
